components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/RemoteFetchConfigurationServiceImpl.java

```diff
@@ -138,8 +138,31 @@ public RemoteFetchConfiguration getRemoteFetchConfiguration(String fetchConfigur
     public void deleteRemoteFetchConfiguration(String fetchConfigurationId)
             throws RemoteFetchCoreException {
 
+        RemoteFetchServiceComponentHolder.getInstance().getRemoteFetchTaskExecutor()
```

Comment:
RemoteFetchTaskExecutor needs to be supplied as a constructor argument or as a setter.
RemoteFetchServiceComponentHolder should not have this.
We need to stick to proper OOP as much as possible